lib/pages/route/route_detail.dart

```diff
@@ -113,13 +113,26 @@ class _DetalleRutaScreenState extends State<DetalleRutaScreen> {
   }
 
   Future<void> _loadRoutePoints() async {
-    final List<LatLng> points = await _fetchRoutePoints();
-    setState(() {
-      routePoints = points;
-    });
-    _updateInterestPoints();
+    if (widget.ruta['local'] == 1) {
+      // Si la ruta es local, cargar los puntos de la base de datos local
+      final pointsData = await db.routes.getPuntosByRutaId(widget.ruta['id'].toString());
+      final List<LatLng> points = pointsData.map((point) => LatLng(point['latitud'], point['longitud'])).toList();
+
+      setState(() {
+        routePoints = points;
+      });
+      _updateInterestPoints(); // Actualiza los puntos de interés
+    } else {
+      // Si la ruta no es local, realizar la solicitud a la API
+      final List<LatLng> points = await _fetchRoutePoints();
+      setState(() {
+        routePoints = points;
+      });
+      _updateInterestPoints(); // Actualiza los puntos de interés
+    }
```

Comment:
Puedes refactorizar esto de esta forma, queda más fácil de entender y hace lo mismo
```suggestion
    late List<LatLng> points;
    if (widget.ruta['local'] == 1) {
      final pointsData = await db.routes.getPuntosByRutaId(widget.ruta['id'].toString());
      points = pointsData.map((point) => LatLng(point['latitud'], point['longitud'])).toList();
    } else {
      points = await _fetchRoutePoints();
    }
    setState(() {
      routePoints = points;
    });
    _updateInterestPoints();
```